pandas/core/algorithms.py

```diff
@@ -1805,6 +1805,8 @@ def map_array(
         return arr.copy()
 
     # we must convert to python types
+    if isinstance(arr.dtype, BaseMaskedDtype):
```

Comment:
Might be better now if `BaseMaskedArray.map` did something like

```
def map(self, mapper, na_action=None):
    return map_array(self.to_numpy(), mapper, na_action=None)
```

Reply:
Could we push this into `BaseMaskedArray.map`?

Reply:
Oh looks like you did this, so this shouldn't be needed anymore?

Reply:
Updated but not sure if this will work, I think I tried this already

lets see what CI says